packages/tre/src/shared/log.ts

```diff
@@ -0,0 +1,123 @@
+import path from "path";
+import { Colorize, dim, green, grey, red, reset, yellow } from "kleur/colors";
+
+const cwd = process.cwd();
+const cwdNodeModules = path.join(cwd, "node_modules");
+
+export enum LogLevel {
+  NONE,
+  ERROR,
+  WARN,
+  INFO,
+  DEBUG,
+  VERBOSE,
+}
+
+const LEVEL_PREFIX: { [key in LogLevel]: string } = {
+  [LogLevel.NONE]: "",
+  [LogLevel.ERROR]: "err",
+  [LogLevel.WARN]: "wrn",
+  [LogLevel.INFO]: "inf",
+  [LogLevel.DEBUG]: "dbg",
+  [LogLevel.VERBOSE]: "vrb",
+};
+
+const LEVEL_COLOUR: { [key in LogLevel]: Colorize } = {
+  [LogLevel.NONE]: reset,
+  [LogLevel.ERROR]: red,
+  [LogLevel.WARN]: yellow,
+  [LogLevel.INFO]: green,
+  [LogLevel.DEBUG]: grey,
+  [LogLevel.VERBOSE]: (input) => dim(grey(input as any)) as any,
+};
```

Comment:
**Non blocking** but Enums are rough on the TS server and compiler and turn into functions at runtime. 
Could use something like an Object literal or Map for the types?